openfecwebapp/views.py

```diff
@@ -46,19 +37,75 @@ def render_table(data_type, results, params):
 
 type_map = {
     'candidates': lambda x: x,
-    'candidate': map_candidate_page_values,
+    'candidate': lambda x: CandidateSchema().dump(x).data,
     'committees': lambda x: x,
-    'committee': map_committee_page_values
+    'committee': lambda x: CommitteeSchema().dump(x).data
 }
 
-def render_page(data_type, c_data):
-    # not handling error at api module because sometimes its ok to 
+
+def render_page(data_type, *args, **kwargs):
+    c_data = args[0]
+    # not handling error at api module because sometimes its ok to
     # not get data back - like with search results
-    if not 'results' in c_data or not c_data['results']:
+    if 'results' not in c_data or not c_data['results']:
         abort(500)
 
-    tmpl_vars = c_data['results'][0]
-    tmpl_vars.update(type_map[data_type](c_data['results'][0]))
-    tmpl_vars.update(add_cmte_financial_data(tmpl_vars, data_type))
+    data = c_data['results'][0]
+
+    if data_type == 'candidate':
+        candidate_schema = CandidateSchema()
+        # candidate fields will be top-level in the template
+        tmpl_vars = candidate_schema.dump(data).data
+        tmpl_vars['election_years'] = load_election_years(data['candidate_id'])
+
+        # process related committees
+        committee_fields = ['committee_id', 'name', 'designation_full', 'designation',
+                            'committee_type_full', 'committee_type', 'url',
+                            'is_primary', 'is_authorized', 'is_joint', 'is_leadership']
+        committees = CommitteeSchema(only=committee_fields, many=True, skip_missing=True, strict=True) \
```

Comment:
Why use `only` here? This data is getting sent to jinja for template rendering, but it's not going to be sent out in the response--is there any cost to sending jinja the entire dictionary?